common/src/main/java/com/datastrato/gravitino/config/ConfigEntry.java

```diff
@@ -136,6 +163,13 @@ public ConfigEntry<Optional<T>> createWithOptional() {
     // Unless explicitly set by the user, null values are not expected to occur.
     conf.setStringConverter(t -> t.map(stringConverter).orElse(null));
     conf.setOptional();
+    conf.setValidator(
+        optionValue -> {
+          if (validator == null) {
+            return;
+          }
+          optionValue.ifPresent(value -> validator.accept(value));
```

Comment:
Can you please optimize the logic here? It is a bit confusing to have both null and optional check here.

Reply:
if optionValue is empty, we will skip the check.

Reply:
I will add some comments here.

Reply:
Fixed.

Reply:
I think you can change like this:

```
Stream.of(Optional.ofNullable(validator), optionValue).allMatch(Optional::isPresent) {
validator.accept(value)
```

Reply:
Fixed.